### ISSUE TITLE
Do not make ProjectGuid a required field in SDK-style sqlproj

### ISSUE THREAD

**Issue by zijchen** [microsoft/azuredatastudio#17847]
- Azure Data Studio Version: 1.34
- OS Version: Win 11
- SQL Database Projects extension: 0.14

Steps to Reproduce:

1. Use SDK style sqlproj by editing project file to include `<Sdk Name="Microsoft.Build.Sql" Version="0.1.1-alpha" />`
2. Remove the `<ProjectGuid>` field as it is unnecessary (even for non-SDK style projects)
3. Database projects extensions throws error when trying to load:
<img width="339" alt="image" src="https://user-images.githubusercontent.com/13544267/145122852-61993231-22b3-4ca8-ad43-5aeb159c3eb2.png">

I think we should just swallow the error if we can't find that field in the sqlproj file.

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by kisantia** [comment]
I looked into this a bit more and things to consider. For building a project, the <ProjectGuid> field isn't necessary. For project references, the <Project> field where the referenced project's guid goes also is not necessary. I didn't find the ProjectGuid used anywhere in DacFx. In SSDT, there are places where there the project guid is read (mostly in schema compare files), but SSDT automatically adds a new project guid to a project when it's opened if it doesn't have one.

example of what's currently added to a sqlproj when a project reference is added:
```
    <ProjectReference Include="..\projectB\projectB.sqlproj">
      <Name>newStyleProject</Name>
      <Project>1dc3e7e1-4119-4c38-9231-177902b69d9b</Project>
      <Private>True</Private>
      <SuppressMissingDependenciesErrors>False</SuppressMissingDependenciesErrors>
    </ProjectReference>
```
but build also works and the referenced project is included if the project guid `<Project>1dc3e7e1-4119-4c38-9231-177902b69d9b</Project>` is removed.

It makes sense to remove showing an error when a project doesn't have a project guid, but the next question is what to do with projects without guids for project references?

A couple options:
1. If a sqlproj doesn't have a project guid, automatically add one like VS. Project references will be added the same as before, with the `<Project>` guid field.
2. Don't add a guid if a project doesn't have one. For project references, don't add the `<Project>` guid field for project references in ADS since it isn't necessary. This would lead to a difference in how project references are added in VS and ADS since VS will still add the `<Project>` guid field.

Any preferences on which option to go with or other ideas @zijchen @Benjin @dzsquared? I'm leaning towards option 1 to be more in line with VS and a project guid will get added anyway if a user opens the project in VS.

**Comment by zijchen** [comment]
I vote option 1 to have parity with VS.
